R/rectangle.R

```diff
@@ -426,15 +434,15 @@ vec_to_wide <- function(x, col, names_sep = NULL) {
   }
 
   if (is.data.frame(x)) {
-    as_tibble(map(x, list))
+    new_data_frame(lapply(x, list))
```

Comment:
The coercion to tibble still happens, but elsewhere?

Reply:
Not directly but this doesn't matter. `vec_to_wide()` is only called from `unnest_wider()` and there the data frame is actually unpacked at the end.

If you are interested here a quick overview of `unnest_wider()`
```r
# make sure that `col` is a list-col of data frames
data[[col]] <- map(data[[col]], vec_to_wide)

# `unchop()` turns `col` into a df-col
data <- unchop(data, any_of(col), keep_empty = TRUE)

# simplify every column of the df-col `col`
data[[col]][] <- map(data[[col]], simplify_col)

# `unpack()` expands the df-col into individual columns
# -> it doesn't matter whether `col` is a data frame or a tibble
unpack(data, any_of(col), names_repair = names_repair)
```